File: Elixir/Hanoi/hanoi.ex
defmodule Hanoi do
  def hanoi(1, from, to, _), do: IO.puts "#{from} -> #{to}"
  def hanoi(n, from, to, via) do
    hanoi(n - 1, from, via, to)
    IO.puts "#{from} -> #{to}"
    hanoi(n - 1, via, to, from)
  end
end

n = elem(Integer.parse(IO.gets ""), 0)
Hanoi.hanoi(n, "a", "b", "c")

File: Ruby/Hanoi/hanoi.rb
def hanoi(n, from, to, via)
  if n > 1
    hanoi(n - 1, from, via, to)
    puts from + " -> " + to
    hanoi(n - 1, via, to, from)
  else
    puts from + " -> " + to
  end
end

n = gets.chomp.to_i
hanoi(n, "a", "b", "c")
